File: lib/cfr/builtins/libs/flutter/material/dateUtils.dart
import 'dart:core' as _fac9;

import 'package:flutter/src/material/date.dart' as _a848;
import 'package:flutter/src/material/material_localizations.dart' as _8728;

import 'package:hydro_sdk/cfr/runtimeSupport.dart' as _36c2;

class VMManagedDateUtils extends _36c2.VMManagedBox<_a848.DateUtils> {
  VMManagedDateUtils(
      {required this.table, required this.vmObject, required this.hydroState})
      : super(
          table: table,
          vmObject: vmObject,
          hydroState: hydroState,
        ) {}

  final _36c2.HydroTable table;

  final _36c2.HydroState hydroState;

  final _a848.DateUtils vmObject;
}

void loadDateUtils(
    {required _36c2.HydroState hydroState, required _36c2.HydroTable table}) {
  table['dateUtilsDateOnly'] = _36c2.makeLuaDartFunc(
      func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
    return [
      _36c2.maybeBoxObject<_fac9.DateTime>(
          object: _a848.DateUtils.dateOnly(
              _36c2.maybeUnBoxAndBuildArgument<_fac9.DateTime, _fac9.dynamic>(
                  luaCallerArguments[1],
                  parentState: hydroState)),
          hydroState: hydroState,
          table: _36c2.HydroTable()),
    ];
  });
  table['dateUtilsDatesOnly'] = _36c2.makeLuaDartFunc(
      func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
    return [
      _36c2.maybeBoxObject<_a848.DateTimeRange>(
          object: _a848.DateUtils.datesOnly(_36c2.maybeUnBoxAndBuildArgument<
              _a848.DateTimeRange,
              _fac9.dynamic>(luaCallerArguments[1], parentState: hydroState)),
          hydroState: hydroState,
          table: _36c2.HydroTable()),
    ];
  });
  table['dateUtilsIsSameDay'] = _36c2.makeLuaDartFunc(
      func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
    return [
      _a848.DateUtils.isSameDay(
          _36c2.maybeUnBoxAndBuildArgument<_fac9.DateTime?, _fac9.dynamic>(
              luaCallerArguments[1],
              parentState: hydroState),
          _36c2.maybeUnBoxAndBuildArgument<_fac9.DateTime?, _fac9.dynamic>(
              luaCallerArguments[2],
              parentState: hydroState)),
    ];
  });
  table['dateUtilsIsSameMonth'] = _36c2.makeLuaDartFunc(
      func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
    return [
      _a848.DateUtils.isSameMonth(
          _36c2.maybeUnBoxAndBuildArgument<_fac9.DateTime?, _fac9.dynamic>(
              luaCallerArguments[1],
              parentState: hydroState),
          _36c2.maybeUnBoxAndBuildArgument<_fac9.DateTime?, _fac9.dynamic>(
              luaCallerArguments[2],
              parentState: hydroState)),
    ];
  });
  table['dateUtilsMonthDelta'] = _36c2.makeLuaDartFunc(
      func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
    return [
      _a848.DateUtils.monthDelta(
          _36c2.maybeUnBoxAndBuildArgument<_fac9.DateTime, _fac9.dynamic>(
              luaCallerArguments[1],
              parentState: hydroState),
          _36c2.maybeUnBoxAndBuildArgument<_fac9.DateTime, _fac9.dynamic>(
              luaCallerArguments[2],
              parentState: hydroState)),
    ];
  });
  table['dateUtilsAddMonthsToMonthDate'] = _36c2.makeLuaDartFunc(
      func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
    return [
      _36c2.maybeBoxObject<_fac9.DateTime>(
          object: _a848.DateUtils.addMonthsToMonthDate(
              _36c2.maybeUnBoxAndBuildArgument<_fac9.DateTime, _fac9.dynamic>(
                  luaCallerArguments[1],
                  parentState: hydroState),
              luaCallerArguments[2]),
          hydroState: hydroState,
          table: _36c2.HydroTable()),
    ];
  });
  table['dateUtilsAddDaysToDate'] = _36c2.makeLuaDartFunc(
      func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
    return [
      _36c2.maybeBoxObject<_fac9.DateTime>(
          object: _a848.DateUtils.addDaysToDate(
              _36c2.maybeUnBoxAndBuildArgument<_fac9.DateTime, _fac9.dynamic>(
                  luaCallerArguments[1],
                  parentState: hydroState),
              luaCallerArguments[2]),
          hydroState: hydroState,
          table: _36c2.HydroTable()),
    ];
  });
  table['dateUtilsFirstDayOffset'] = _36c2.makeLuaDartFunc(
      func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
    return [
      _a848.DateUtils.firstDayOffset(
          luaCallerArguments[1],
          luaCallerArguments[2],
          _36c2.maybeUnBoxAndBuildArgument<_8728.MaterialLocalizations,
              _fac9.dynamic>(luaCallerArguments[3], parentState: hydroState)),
    ];
  });
  table['dateUtilsGetDaysInMonth'] = _36c2.makeLuaDartFunc(
      func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
    return [
      _a848.DateUtils.getDaysInMonth(
          luaCallerArguments[1], luaCallerArguments[2]),
    ];
  });
  _36c2.registerBoxer<_a848.DateUtils>(boxer: (
      {required _a848.DateUtils vmObject,
      required _36c2.HydroState hydroState,
      required _36c2.HydroTable table}) {
    return VMManagedDateUtils(
        vmObject: vmObject, hydroState: hydroState, table: table);
  });
}

File: runtime/flutter/material/dateUtils.ts
import { IDateTime } from "../../dart/core/dateTime";
import { IDateTimeRange } from "./dateTimeRange";
import { IMaterialLocalizations } from "./materialLocalizations";
declare const flutter: {
    material: {
        dateUtilsDateOnly: (date: IDateTime) => IDateTime;
        dateUtilsDatesOnly: (range: IDateTimeRange) => IDateTimeRange;
        dateUtilsIsSameDay: (
            dateA?: IDateTime | undefined,
            dateB?: IDateTime | undefined
        ) => boolean;
        dateUtilsIsSameMonth: (
            dateA?: IDateTime | undefined,
            dateB?: IDateTime | undefined
        ) => boolean;
        dateUtilsMonthDelta: (
            startDate: IDateTime,
            endDate: IDateTime
        ) => number;
        dateUtilsAddMonthsToMonthDate: (
            monthDate: IDateTime,
            monthsToAdd: number
        ) => IDateTime;
        dateUtilsAddDaysToDate: (date: IDateTime, days: number) => IDateTime;
        dateUtilsFirstDayOffset: (
            year: number,
            month: number,
            localizations: IMaterialLocalizations
        ) => number;
        dateUtilsGetDaysInMonth: (year: number, month: number) => number;
    };
};
export interface IDateUtils {}
export class DateUtils {
    public static dateOnly(date: IDateTime): IDateTime {
        return flutter.material.dateUtilsDateOnly(date);
    }
    public static datesOnly(range: IDateTimeRange): IDateTimeRange {
        return flutter.material.dateUtilsDatesOnly(range);
    }
    public static isSameDay(
        dateA?: IDateTime | undefined,
        dateB?: IDateTime | undefined
    ): boolean {
        return flutter.material.dateUtilsIsSameDay(dateA, dateB);
    }
    public static isSameMonth(
        dateA?: IDateTime | undefined,
        dateB?: IDateTime | undefined
    ): boolean {
        return flutter.material.dateUtilsIsSameMonth(dateA, dateB);
    }
    public static monthDelta(startDate: IDateTime, endDate: IDateTime): number {
        return flutter.material.dateUtilsMonthDelta(startDate, endDate);
    }
    public static addMonthsToMonthDate(
        monthDate: IDateTime,
        monthsToAdd: number
    ): IDateTime {
        return flutter.material.dateUtilsAddMonthsToMonthDate(
            monthDate,
            monthsToAdd
        );
    }
    public static addDaysToDate(date: IDateTime, days: number): IDateTime {
        return flutter.material.dateUtilsAddDaysToDate(date, days);
    }
    public static firstDayOffset(
        year: number,
        month: number,
        localizations: IMaterialLocalizations
    ): number {
        return flutter.material.dateUtilsFirstDayOffset(
            year,
            month,
            localizations
        );
    }
    public static getDaysInMonth(year: number, month: number): number {
        return flutter.material.dateUtilsGetDaysInMonth(year, month);
    }
}
